Source/WebCore/dom/ScriptExecutionContext.h

```diff
@@ -179,6 +180,8 @@ class ScriptExecutionContext : public SecurityContext, public TimerAlignment {
 
     URL currentSourceURL() const;
 
+    OptionSet<FilterRenderingMode> preferredFilterRenderingModes() const;
```

Comment:
>For example a free function in Filter.h, or static function in Filter ?
>OptionSet<FilterRenderingMode> resolveFilterRenderingMode(const SettingsValues&)

This is will be layering violation. The code in the `platform/` directory should not use code from the `dom/` directory.

I agree putting this function in `ScriptExecutionContext` is not ideal. But since the `filter` of the canvas is applied through the JavaScript, it was convenient to have this function in `ScriptExecutionContext`.

Currently this function is in the `Page` class which is not ideal either and it does not work for Offscreen canvas. So I really do not know where to put it. Maybe we should have a separate file for this function.

Reply:
I reverted this change. `preferredFilterRenderingModes()` is still a method of `Page`.